pkg/api/defaults-kubelet.go

```diff
@@ -133,6 +133,18 @@ func (cs *ContainerService) setKubeletConfig() {
 		}
 	}
 
+	if isUpgrade && common.IsKubernetesVersionGe(o.OrchestratorVersion, "1.14.0") {
+		hasSupportPodPidsLimitFeatureGate := strings.Contains(o.KubernetesConfig.KubeletConfig["--feature-gates"], "SupportPodPidsLimit=true")
+		podMaxPids, _ := strconv.Atoi(o.KubernetesConfig.KubeletConfig["--pod-max-pids"])
+		if podMaxPids > 0 {
+			// If we don't have an explicit SupportPodPidsLimit=true, disable --pod-max-pids by setting to -1
+			// To prevent older clusters from inheriting SupportPodPidsLimit=true implicitly starting w/ 1.14.0
+			if !hasSupportPodPidsLimitFeatureGate {
+				o.KubernetesConfig.KubeletConfig["--pod-max-pids"] = strconv.Itoa(-1)
```

Comment:
isn't the default value already -1? https://github.com/Azure/aks-engine/blob/master/pkg/api/defaults-kubelet.go#L78

Reply:
Oh I think I understand what's going on... we aren't overriding the default value during upgrades and this default was previously 100

Reply:
Yeah, it's basically a timebomb. A config value of "100", that, if enabled, would demonstrably change the operational reality of cluster behaviors. Then it gets enabled (in 1.14.1).

This is why we did this:

https://github.com/Azure/aks-engine/pull/1508

We just didn't address the back-compat ramifications.

Reply:
Whoops I meant https://github.com/Azure/aks-engine/pull/1126 changed the value to -1